airflow_dags/wikipedia/ngrams/dolt_load.py

```diff
@@ -0,0 +1,250 @@
+import re
+import subprocess
+import time
+import logging
+import requests
+import nltk
+from collections import defaultdict, Counter
+from os import path
+from pathlib import Path
+from doltpy.etl import get_df_table_writer, get_dolt_loader, get_branch_creator, dolthub_loader
+import pandas as pd
+from typing import Callable
+from functools import reduce, partial
+import operator
+import string
+from itertools import islice
+
+logger = logging.getLogger(__name__)
+
+CURR_DIR = path.dirname(path.abspath(__file__))
+WIKIEXTRACTOR_PATH = path.join(Path(CURR_DIR).parent, 'wikiextractor/WikiExtractor.py')
+
+NGRAM_DICTS = {
+    'unigram': defaultdict(lambda: defaultdict(int)),
+    'bigram': defaultdict(lambda: defaultdict(int)),
+    'trigram': defaultdict(lambda: defaultdict(int)),
+}
+
+LINE_TRANS = str.maketrans('–’', "-\'")
+NOT_PUNCTUATION = re.compile(r'.*[a-zA-Z0-9].*')
+NON_WORD_PUNCT = re.sub(r'[.\'&-]', '', string.punctuation)
+
+
+def fetch_data(dump_target: str):
+    bz2_file_name = 'enwiki-{}-pages-articles-multistream.xml.bz2'.format(dump_target)
+    dump_url = 'https://dumps.wikimedia.your.org/enwiki/{}/{}'.format(dump_target, bz2_file_name)
+    dump_path = path.join(CURR_DIR, bz2_file_name)
+
+    if path.exists(dump_path):
+        logging.info('Using existing XML dump.')
+    else:
```

Comment:
Will remove this, but don't want to download dump every time I'm testing locally